### PR TITLE
Add GitHub Actions workflows, update demo_spec.js and commands

### PR DESCRIPTION
**Pull request by avivbenchorin**
### Description

This PR aims to add CI capabilities to the Monterey project by adding two GitHub Actions workflows, with one setting up OpenSearch and OpenSearch Dashboards by building them from source, and the other setting them up using downloaded snapshots. Both workflows prepare OpenSearch and OpenSearch Dashboards, checkout the Monterey project's repository, and then use the official [Cypress GitHub Action](https://github.com/marketplace/actions/cypress-io) to run all test files in a headless Chrome browser. After the end of the test runs, the resulting artifacts, such as videos and snapshots, as uploaded to GitHub and accessible in the summary of the These workflows run automatically upon pushes or pull requests to main or any dev branches. The results of the runs can be found [here](https://github.com/AvivBenchorin/monterey/actions).

There currently exists some issues with the proposed CI implementation using GitHub Actions. First, the Cypress tests run substantially slower on GitHub Actions compared to locally (3.5 to 5 minutes vs 2 to 2.5 minutes locally). Second, the recorded videos from the test run tend to freeze for long periods of time, sometimes making it impossible to follow the events of the tests and reducing the effectiveness of the videos for debugging purposes. Both of these concerns may be caused by Cypress over-stressing the resources of the [GitHub-hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources). A future PR will try to improve performance by reducing the window size used when running the tests (as described [here](https://docs.cypress.io/api/plugins/browser-launch-api#Set-screen-size-when-running-headless)).

The following videos come from [this](https://github.com/AvivBenchorin/monterey/actions/runs/1126063700) test run, and showcase the extended runtime duration and frozen videos:

https://user-images.githubusercontent.com/47436232/129293667-473b446d-2ebf-4884-9b93-ab67420f025d.mp4

https://user-images.githubusercontent.com/47436232/129293673-a3ae1fa3-40bd-4bd2-8d4c-33a6b77a45f5.mp4

This commit also makes other minor improvements to the project, such as updating the `demo_spec.js` and `dashboard_filtering_spec.js` test files, and reorganizing the Cypress custom commands into multiple files. These improvements are:

1. Updating `demo_spec.js` to utilize the recently developed Cypress custom commands (such as when applying a filter to the dashboard), as well as redesigning the method of initializing the sample data to reduce inconsistency when running the test.
2. Replacing the "magic number" for the number of retries that the `addFilterRetryFull` and `addFilterRetrySelection` functions can do with a parameter in the function. The parameter has a default value of 3.
3. Adding functionality to the `addDashboardPanels` function to close the toast popups that appear after adding each saved visualization.
4. Restructuring the Cypress custom commands into three files (`common_commands.js`, `dashboard_commands.js`, and `network_commands.js`) to make it easier to go through the files and improve maintainability. Some of the custom commands have been renamed to reflect the location changes; for example, `saveVisualization` was changed to `saveDashboardVisualization` after being moved to `dashboard_commands.js` to reduce the risk of confusion in the future.

Signed-off-by: Aviv Benchorin <benchori@amazon.com>

### Issues Resolved

Partially addresses: #6 

### Check List

- [ ] New functionality includes testing. 
    - [x] All tests pass
- [ ] New functionality has been documented. 
    - [x] New functionality has javadoc added 
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: cypress/support/dashboard_commands.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        })
+        iteratePages()
+      }
+    })
+  }
+  cy.get('[data-test-subj="dashboardAddPanelButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderItemList"]').should('be.visible')
+
+  cy.get('[data-test-subj="savedObjectFinderFilterButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderFilter-' + type + '"]').should('be.visible').click()
```
Comment (seraphjiang) [inline cypress/support/dashboard_commands.js]: use template literals

` `[data-test-subj="savedObjectFinderFilter-${type}"]` `

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: cypress/support/dashboard_commands.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  cy.get('[data-test-subj="savedObjectFinderFilterButton"]').should('be.visible').click()
+
+  cy.get('[data-test-subj="savedObjectFinderSearchInput"]').should('be.visible').type(keyword)
+
+  cy.get('[data-test-subj^="savedObjectTitle' + keyword.replace(/\s+/g, '-') + '"]').should('be.visible').each(($button) => {
```
Comment (seraphjiang) [inline cypress/support/dashboard_commands.js]: refactory below piece code to a method for readability

``keyword.replace(/\s+/g, '-') + '"]'``

File: cypress/support/common_commands.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
+/**
+ * Sets the page data range
```
Comment (seraphjiang) [inline cypress/support/common_commands.js]: date range?

File: cypress/support/network_commands.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    const strSplit = str.split('\n\n')
+    cy.wrap(strSplit).each((element) => {
+      const json = JSON.parse(element)
+      if (json.type === 'index') {
+        const index = json.value.index
+        const settings = json.value.settings
+        const mappings = json.value.mappings
+        const aliases = json.value.aliases
+        const body = { settings, mappings, aliases }
+        cy.request({ method: 'PUT', url: hostname + ':' + port + '/' + index, body: body, failOnStatusCode: false }).then((response) => {
```
Comment (seraphjiang) [inline cypress/support/network_commands.js]: format and pretty code/json

**Review by seraphjiang — APPROVED** [review]
LGTM
Add a few trivial feedback, you may address in future PR

File: cypress/support/dashboard_commands.js (hunk 1 above)
Comment (avivbenchorin) [reply cypress/support/dashboard_commands.js]: These changes have been applied, the project should now be fully refactored to utilize template literals where appropriate.

File: cypress/support/dashboard_commands.js (hunk 2 above)
Comment (avivbenchorin) [reply cypress/support/dashboard_commands.js]: I now refactored the specified code to its own constant function (`replaceSpacesWithDashes`).

File: cypress/support/common_commands.js (hunk 3 above)
Comment (avivbenchorin) [reply cypress/support/common_commands.js]: Thank you for the catch, it has been corrected!

File: cypress/support/network_commands.js (hunk 4 above)
Comment (avivbenchorin) [reply cypress/support/network_commands.js]: Would you be able to clarify in what way the format should be changed? Is it changing the use of the constant variables and/or the format of the `cy.request` call?

File: .github/workflows/cypress-workflow-snapshot-based.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      - name: Get and run OpenSearch
+        run: |
+          wget https://artifacts.opensearch.org/snapshots/core/opensearch/1.0.0-SNAPSHOT/opensearch-min-1.0.0-SNAPSHOT-linux-x64-latest.tar.gz
```
Comment (tianleh) [inline .github/workflows/cypress-workflow-snapshot-based.yml:27]: nonblocking: create a variable for opensearch version "1.0.0" to make it easy for future releases. In this way, there is no need to modify the "1.0.0" in multiple places.

**Review by tianleh — COMMENTED** [review]
noblocking: great effort on the refactoring part. Better to have separate PRs for the refactoring and the new Cypress workflow. Just for future references.

**Review by tianleh — APPROVED** [review]

File: .github/workflows/cypress-workflow-snapshot-based.yml (hunk 5 above)
Comment (avivbenchorin) [reply .github/workflows/cypress-workflow-snapshot-based.yml:27]: That makes sense, I can make that change in a future PR!

**Pull request merged**